lib/puppet/parser/functions/ensure_php_packages.rb

```diff
@@ -0,0 +1,33 @@
+module Puppet::Parser::Functions
+  newfunction(:ensure_php_packages, :type => :statement) do |arguments|
+    if arguments.size > 3 or arguments.size == 0
+      raise(Puppet::ParseError, "ensure_php_packages(): Wrong number of arguments given (#{arguments.size} for 2 or 3)")
+    elsif arguments.size == 3 and !arguments[2].is_a?(Hash)
+      raise(Puppet::ParseError, 'ensure_php_packages(): Requires third argument to be a Hash')
+    end
+
+    packages = arguments[0]
+    versions = arguments[1]
+
+    unless packages.is_a?(Array) && versions.is_a?(Array)
+      raise(Puppet::ParseError, "ensure_php_packages(): Requires 2 arrays")
+    end
+
+    if arguments[2]
+      defaults = { "ensure" => "present" }.merge(arguments[2])
+    else
+      defaults = { "ensure" => "present" }
+    end
+
+    versions.each do |version|
+      packages.each do |package|
+        # skip mcrypt on 7.2 and greater, the package was removed.
```

Comment:
A bit redundant. Probably just say "Package is now part of core"?

Reply:
it isn't part of core, it got removed. I'll just ditch the comment